FSharpLu.Json/Compact.fs

```diff
@@ -10,18 +13,33 @@ open Microsoft.FSharp.Reflection
 type CompactUnionJsonConverter() =
     inherit Newtonsoft.Json.JsonConverter()
 
+    let processWithContractResolver (serializer:JsonSerializer) (name:string) =
+        let toPascal (name:string) =
+            if System.Char.IsUpper (name, 0) then name
+            else string(System.Char.ToUpper name.[0]) + name.Substring(1)  
+        let toCamel (name:string) =
+            if System.Char.IsLower (name, 0) then name
+            else string(System.Char.ToLower name.[0]) + name.Substring(1)    
+        match serializer.ContractResolver with 
+        | :? CamelCasePropertyNamesContractResolver -> toCamel name
+        | :? DefaultContractResolver -> toPascal name
+        | _ -> name
+
     let SomeFieldIdentifier = "Some"
 
+    let stringEq (a:string) (b:string) =
+        a.Equals(b, System.StringComparison.InvariantCultureIgnoreCase)
+
     let isOptionType (t:System.Type) =
        t.IsGenericType && t.GetGenericTypeDefinition() = typedefof<option<_>>
 
     /// Determine if a given type has a field named 'Some' which would cause
     /// ambiguity if nested under an option type without being boxed
     let hasFieldNamedSome (t:System.Type) =
         isOptionType t // the option type itself has a 'Some' field
-        || (FSharpType.IsRecord t && FSharpType.GetRecordFields t |> Seq.exists (fun r -> r.Name = SomeFieldIdentifier))
-        || (FSharpType.IsUnion t && FSharpType.GetUnionCases t |> Seq.exists (fun r -> r.Name = SomeFieldIdentifier))
-        || (FSharpType.IsUnion t && FSharpType.GetUnionCases t |> Seq.exists (fun r -> r.Name = SomeFieldIdentifier))
+        || (FSharpType.IsRecord t && FSharpType.GetRecordFields t |> Seq.exists (fun r -> stringEq r.Name SomeFieldIdentifier))
+        || (FSharpType.IsUnion t && FSharpType.GetUnionCases t |> Seq.exists (fun r -> stringEq r.Name SomeFieldIdentifier))
+        || (FSharpType.IsUnion t && FSharpType.GetUnionCases t |> Seq.exists (fun r -> stringEq r.Name SomeFieldIdentifier))
```

Comment:
Just realize that the last two lines are identical, you can remove the last line.